### PR TITLE
Added more generic run_replicates method and notebook to demo it.

### PR DESCRIPTION
**Pull request by MichaelClerx**
Hi @ben18785 , @chonlei , @rcw5890 !

It struck me that the run_replicates_x methods can all be combined if we just re-use the keys already in the FT results? (Which means having 'mean-ess' instead of 'ess' but is otherwise the same).

Has the added advantage of _potentially_ allowing parallelisation, although this can get a bit messy in notebooks

Notebook here shows the results are the same. If you like it, I can change the remaining notebooks to all use this function, and remove the other run_replicates_x versions?

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -3,9 +3,8 @@
...
 import numpy as np
 import pandas as pd
-import matplotlib.pyplot as plt
-from scipy.stats import norm
-from statsmodels.distributions.empirical_distribution import ECDF
+
+import pints
 
 
 def run_replicates(iterations, n_replicates, test):
```
Comment (ben18785) [inline utils.py]: Do we still need this one after adding `run_replicates2`?

**Review by ben18785 — APPROVED** [review]
Thanks @MichaelClerx -- looks great. My only thought was, should we remove the redundant `run_replicates` function?

File: utils.py (hunk 1 above)
Comment (MichaelClerx) [reply utils.py]: I'm happy to rewrite the other tests to all use `run_replicates`, if you like

Comment (ben18785) [reply utils.py]: Yes, aren't all methods (e.g. `run_replicates_distance`, `run_replicates_annulus`) now handled by your method? As in, I'm not sure I see what rewriting would need to be done?

Comment (MichaelClerx) [reply utils.py]: I wasn't sure if that was the case so I didn't replace the other _x methods yet. Doing a PR for that now though!

**Comment by MichaelClerx** [comment]
Updated all notebooks, removed the demo notebook, removed the redundant methods in utils.

**Pull request merged**